File: python/busqueda_lineal_pra.py
def busqueda_lineal(lista, nombre):
    for estudiante in lista:
        if estudiante["nombre"] == nombre:
            return estudiante

    return "No encontrado"


estudiantes = [
    {"nombre": "Ana"},
    {"nombre": "Luis"}
]

print(busqueda_lineal(estudiantes, "Luis"))

File: js/busqueda_lineal_pra.js
function busquedaLineal(lista, nombre) {
    for (let estudiante of lista) {
        if (estudiante.nombre === nombre) {
            return estudiante;
        }
    }

    return "No encontrado";
}

let estudiantes = [
    {nombre: "Ana"},
    {nombre: "Luis"}
];

console.log(busquedaLineal(estudiantes, "Luis"));